src/aiida/engine/daemon/client.py

```diff
@@ -155,32 +155,32 @@ def virtualenv(self) -> str | None:
         return VIRTUALENV
 
     @property
-    def circus_log_file(self) -> str:
-        return self.profile.filepaths['circus']['log']
+    def circus_log_file(self) -> pathlib.Path:
+        return self.profile.circus_log_file
 
     @property
-    def circus_pid_file(self) -> str:
-        return self.profile.filepaths['circus']['pid']
+    def circus_pid_file(self) -> pathlib.Path:
+        return self.profile.circus_pid_file
 
     @property
-    def circus_port_file(self) -> str:
-        return self.profile.filepaths['circus']['port']
+    def circus_port_file(self) -> pathlib.Path:
+        return self.profile.circus_port_file
 
     @property
-    def circus_socket_file(self) -> str:
-        return self.profile.filepaths['circus']['socket']['file']
+    def circus_socket_file(self) -> pathlib.Path:
+        return self.profile.circus_socket_endpoints['file']
 
     @property
-    def circus_socket_endpoints(self) -> dict[str, str]:
-        return self.profile.filepaths['circus']['socket']
+    def circus_socket_endpoints(self) -> dict[str, pathlib.Path]:
+        return self.profile.circus_socket_endpoints
 
     @property
-    def daemon_log_file(self) -> str:
-        return self.profile.filepaths['daemon']['log']
+    def daemon_log_file(self) -> pathlib.Path:
+        return self.profile.daemon_log_file
 
     @property
-    def daemon_pid_file(self) -> str:
```

Comment:
Since it is python, I don't regard it as a "breaking change". The mypy might be unhappy if you already use types on downsteram plugins. But functionality side it won't break anything.

Reply:
Sorry, I think I am not correct, the function return type changed actually. 
Then my argument would be, this is the function I think should not exposed as public API.